src/main/java/at/hannibal2/skyhanni/data/TitleManager.kt

```diff
@@ -3,66 +3,126 @@ package at.hannibal2.skyhanni.data
 import at.hannibal2.skyhanni.api.event.HandleEvent
 import at.hannibal2.skyhanni.config.commands.CommandCategory
 import at.hannibal2.skyhanni.config.commands.CommandRegistrationEvent
+import at.hannibal2.skyhanni.config.storage.ResettableStorageSet
 import at.hannibal2.skyhanni.events.GuiRenderEvent
 import at.hannibal2.skyhanni.events.ProfileJoinEvent
 import at.hannibal2.skyhanni.events.minecraft.SkyHanniTickEvent
 import at.hannibal2.skyhanni.skyhannimodule.SkyHanniModule
 import at.hannibal2.skyhanni.utils.ChatUtils
+import at.hannibal2.skyhanni.utils.CollectionUtils
+import at.hannibal2.skyhanni.utils.CollectionUtils.enumMapOf
+import at.hannibal2.skyhanni.utils.ColorUtils
+import at.hannibal2.skyhanni.utils.RenderUtils
 import at.hannibal2.skyhanni.utils.SimpleTimeMark
 import at.hannibal2.skyhanni.utils.compat.GuiScreenUtils
-import io.github.notenoughupdates.moulconfig.internal.TextRenderUtils
+import at.hannibal2.skyhanni.utils.renderables.Renderable
+import at.hannibal2.skyhanni.utils.renderables.RenderableUtils.renderXYAligned
 import net.minecraft.client.Minecraft
+import net.minecraft.client.gui.inventory.GuiContainer
 import net.minecraft.client.renderer.GlStateManager
 import org.lwjgl.opengl.GL11
-import java.util.LinkedList
-import java.util.Queue
+import kotlin.math.min
 import kotlin.time.Duration
 import kotlin.time.Duration.Companion.seconds
 
 @SkyHanniModule
 object TitleManager {
 
-    private var currentText = ""
-    private var display = ""
-    private var endTime = SimpleTimeMark.farPast()
-    private var heightModifier = 1.8
-    private var fontSizeModifier = 4f
+    private val titleLocationQueues: MutableMap<TitleLocation, CollectionUtils.OrderedQueue<TitleData>> = enumMapOf()
+    private val currentTitles: MutableMap<TitleLocation, TitleData?> = enumMapOf()
+
+    private data class TitleData(
+        var text: String = "",
+        var duration: Duration = 0.seconds,
```

Comment:
do we need title data to be a data class? if not, we can remove duration from class member and only have it be a parameter in constructor

Reply:
duration needs to be persistent, so that if a title is "importantly sent", i.e., with a higher weight, and pushes the current title to the front of the queue, when it's de-queued again, it has that duration still.